### PR TITLE
avoid failing when lgetxattr has no data to give

### PR DESCRIPTION
**Pull request by hallyn**

### CODE REVIEW

**Comment by hallyn** [comment]
See https://github.com/project-stacker/stacker/pull/586

File: oci/layer/tar_generate.go [hunk 1]
```diff
@@ -219,10 +219,14 @@ func (tg *tarGenerator) AddFile(name, path string) error {
...
 		if err != nil {
-			// XXX: I'm not sure if we're unprivileged whether Lgetxattr can
-			//      fail with EPERM. If it can, we should ignore it (like when
-			//      we try to clear xattrs).
-			return errors.Wrapf(err, "get xattr: %s", name)
+			v := errors.Cause(err)
+			log.Debugf("failure reading xattr from list on %q: %q", name, v)
+			if v != unix.EOPNOTSUPP && v != unix.ENODATA && v != unix.EPERM {
+				// XXX: I'm not sure if we're unprivileged whether Lgetxattr can
+				//      fail with EPERM. If it can, we should ignore it (like when
```
Comment (tych0) [inline oci/layer/tar_generate.go:226]: this comment is obsolete with your change, right? (you are ignoring it now?)

Comment (hallyn) [reply oci/layer/tar_generate.go:226]: lol - yes.  in fact I didn't have EPERM in there at first, and the comment convinced me to do it.  Then I left it anyway :)

This is IMO the most controversial part, still: if we get EPERM do we want to have an error, or ignore it?

Comment (hallyn) [reply oci/layer/tar_generate.go:226]: Sounds good - updated the commit to do that.  Thanks.

**Review by tych0 — APPROVED** [review]

**Comment by tych0** [comment]
if you rebase hopefully CI will pass.

**Pull request merged**